### PR TITLE
[palllet-ibc] Introduce config option to enforce minimum connection delay period

### PR DESCRIPTION
**Pull request by seunlanlege**

### CODE REVIEW

File: contracts/pallet-ibc/README.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ parameter_types! {
     pub const ExpectedBlockTime: u64 = 12000;
     pub const SpamProtectionDeposit: Balances = 10000;  
     pub const RelayChainId: light_client_commomn::RelayChain = light_client_commomn::RelayChain::Rococo;
-    pub const NativeAssetId: AssetId = 1;
+    pub const NativeAssetId: AssetId = 1
+    pub const MinimumConnectionDelay: u64 = 600; // 5 minutes
```
Comment (blasrodri) [inline contracts/pallet-ibc/README.md:31]: Maybe you want 300 if it's 5 mins?

File: contracts/pallet-ibc/src/benchmarks/tendermint_benchmark_utils.rs [hunk 2]
```diff
@@ -181,7 +181,7 @@ where
 		Some(ConnectionId::new(1)),
 		commitment_prefix.clone(),
 	);
-	let delay_period = core::time::Duration::from_nanos(1000);
+	let delay_period = core::time::Duration::from_secs(1000);
```
Comment (blasrodri) [inline contracts/pallet-ibc/src/benchmarks/tendermint_benchmark_utils.rs:184]: do we want a delay of > 15 mins?

Comment (seunlanlege) [reply contracts/pallet-ibc/src/benchmarks/tendermint_benchmark_utils.rs:184]: this is just for tests

File: contracts/pallet-ibc/README.md (hunk 1 above)
Comment (seunlanlege) [reply contracts/pallet-ibc/README.md:31]: yeah will fix

File: ibc/modules/src/core/ics03_connection/context.rs [hunk 3]
```diff
@@ -16,9 +16,14 @@ use crate::{
 	prelude::*,
 	Height,
 };
+use core::time::Duration;
 
 /// A context supplying all the necessary read-only dependencies for processing any `ConnectionMsg`.
 pub trait ConnectionReader {
+	/// Minimum connection delay period for ibc connections that can be created or accepted.
```
Comment (blasrodri) [inline ibc/modules/src/core/ics03_connection/context.rs:23]:
```suggestion
	/// Minimum connection delay period for IBC connections that can be created or accepted.
```

File: utils/parachain-node/runtime/src/lib.rs [hunk 4]
```diff
@@ -515,6 +515,7 @@ parameter_types! {
 	pub const RelayChainId: RelayChain = RelayChain::Rococo;
 	pub const SpamProtectionDeposit: Balance = 1_000_000_000_000;
 	pub const NativeAssetId: AssetId = 1;
+	pub const MinimumConnectionDelay: u64 = 0; // well we don't need the security tbh.
```
Comment (blasrodri) [inline utils/parachain-node/runtime/src/lib.rs:518]: > we don't need the security?
lol, how's that?

Comment (seunlanlege) [reply utils/parachain-node/runtime/src/lib.rs:518]: this node only exists for tests as well

**Review by blasrodri — DISMISSED** [review]

File: contracts/pallet-ibc/README.md (hunk 1 above)
Comment (seunlanlege) [inline contracts/pallet-ibc/README.md:31]:
```suggestion
    pub const MinimumConnectionDelay: u64 = 300; // 5 minutes
```

**Review by Wizdave97 — DISMISSED** [review]

**Review by blasrodri — DISMISSED** [review]

**Review by Wizdave97 — APPROVED** [review]

**Pull request merged**